components/script/dom/document.rs

```diff
@@ -3809,9 +3809,19 @@ impl DocumentMethods for Document {
         }
 
         // Step 3.
+        if let Some(fragment) = node.downcast::<DocumentFragment>() {
+            if fragment.get_host().is_some() {
+                // Spec text doesn't actually say what to return,
+                // but IDL doesn't allow null, and
+                // returning the node itself seems to be what's expected
```

Comment:
I went to open a spec issue about this and found https://github.com/whatwg/dom/issues/813.